docs/resources/nat_private_dnat_rule.md

```diff
@@ -0,0 +1,159 @@
+variable "transit_ip_id" {}
+
+resource "huaweicloud_compute_instance" "test" {
+  ...
+}
+
+resource "huaweicloud_nat_private_dnat_rule" "test" {
+  gateway_id            = var.gateway_id
+  protocol              = "tcp"
+  transit_ip_id         = var.transit_ip_id
+  transit_service_port  = 1000
+  backend_interface_id  = huaweicloud_compute_instance.test.network[0].port
+  internal_service_port = 2000
+}
+```
+
+### DNAT rules forwarded with ELB loadbalancer as the backend
+
+```hcl
+variable "network_id" {}
+variable "gateway_id" {}
+variable "transit_ip_id" {}
+
+resource "huaweicloud_elb_loadbalancer" "test" {
+  ...
+}
+
+data "huaweicloud_networking_port" "test" {
+  network_id = var.network_id
+  fixed_ip   = huaweicloud_elb_loadbalancer.test.ipv4_address
+}
+
+resource "huaweicloud_nat_private_dnat_rule" "test" {
+  gateway_id            = var.gateway_id
+  protocol              = "tcp"
+  transit_ip_id         = var.transit_ip_id
+  transit_service_port  = 1000
+  backend_interface_id  = data.huaweicloud_networking_port.test.id
+  internal_service_port = 2000
+}
+```
+
+### DNAT rules forwarded with VIP as the backend
+
+```hcl
+variable "network_id" {}
+variable "gateway_id" {}
+variable "transit_ip_id" {}
+
+resource "huaweicloud_networking_vip" "test" {
+  network_id = var.network_id
+}
+
+resource "huaweicloud_nat_private_dnat_rule" "test" {
+  gateway_id            = var.gateway_id
+  protocol              = "tcp"
+  transit_ip_id         = var.transit_ip_id
+  transit_service_port  = 1000
+  backend_interface_id  = huaweicloud_networking_vip.test.id
+  internal_service_port = 2000
+}
+```
+
+### DNAT rules forwarded with a custom private IP address as the backend
+
+```hcl
+variable "gateway_id" {}
+variable "transit_ip_id" {}
+
+resource "huaweicloud_nat_private_dnat_rule" "test" {
+  gateway_id            = var.gateway_id
+  protocol              = "tcp"
+  transit_ip_id         = var.transit_ip_id
+  transit_service_port  = 1000
+  backend_private_ip    = "172.168.0.69"
+  internal_service_port = 2000
+}
+```
+
+### DNAT rules for all ports
+
+```hcl
+variable "gateway_id" {}
+variable "transit_ip_id" {}
+
+resource "huaweicloud_nat_private_dnat_rule" "test" {
+  gateway_id         = var.gateway_id
+  protocol           = "any"
+  transit_ip_id      = var.transit_ip_id
+  backend_private_ip = "172.168.0.69"
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `region` - (Optional, String, ForceNew) Specifies the region where the DNAT rule is located.  
+  If omitted, the provider-level region will be used. Changing this will create a new resource.
+
+* `gateway_id` - (Required, String, ForceNew) Specifies the private NAT gateway ID to which the DNAT rule belongs.  
+  Changing this will create a new resource.
+
+* `transit_ip_id` - (Required, String) Specifies the ID of the transit IP for private NAT.
+
+* `transit_service_port` - (Optional, Int) Specifies the port of the transit IP.  
```

Comment:
add a description of the default, for example `defaults to 0, which indicates any`

Reply:
done.